### PR TITLE
Fix Generic CLI Client Error Handling and Mode Default

### PR DESCRIPTION
**Pull request by danintel**
- Add default to `-m`/`--mode` command line option
- Change `logger.info` or `logger.warn` to `logger.error` if before `sys.exit`

Signed-off-by: danintel <daniel.anderson@intel.com>

### CODE REVIEW

**Comment by danintel** [comment]
A symptom of the bug is this error message
`Address needs to be passed with mode option`

File: examples/apps/generic_client/generic_client.py [hunk 1]
```diff
@@ -66,7 +66,8 @@ def ParseCommandLine(args) :
 		help="an address (hex string) of the smart contract (e.g. Worker registry listing)",
 		type=str)
 	parser.add_argument("-m", "--mode",
-		help="should be one of listing or registry",
+		help="should be one of listing or registry (default)",
```
Comment (arsulegai) [inline examples/apps/generic_client/generic_client.py]: I found an answer on StackOverflow, can that be used here to reduce boilerplate code?
https://stackoverflow.com/questions/15836713/allowing-specific-values-for-an-argparse-argument

Comment (danintel) [reply examples/apps/generic_client/generic_client.py]: Thanks.I added `               choices={"registry", "listing"}, for `--mode`

File: examples/apps/generic_client/generic_client.py [hunk 2]
```diff
@@ -119,16 +121,18 @@ def ParseCommandLine(args) :
 			sys.exit(-1)
 		else:
 			logger.error("Mode should be either registry or listing")
+			sys.exit(-1)
 
 	else:
 		if not mode:
 			logger.error("Address needs to be passed with mode option")
+			sys.exit(-1)
```
Comment (manojgop) [inline examples/apps/generic_client/generic_client.py]: Will the mode be set to default "registry" if not explicitly set with the above changes. If so, mode can never be None. Hence "if not mode" condition check is not required I think.

Comment (danintel) [reply examples/apps/generic_client/generic_client.py]: I removed the code (lines 122-129 above, both `else` clauses)

Comment (danintel) [reply examples/apps/generic_client/generic_client.py]: Correct.  I removed the `if not mode` check.

**Pull request merged**